compiler/shared_ast/interpreter.ml

```diff
@@ -497,7 +499,17 @@ let rec runtime_to_val :
     in
     let e = runtime_to_val eval_expr ctx m ty (Obj.field o 0) in
     EInj { name; cons; e }, m
-  | TOption _ty -> assert false
+  | TOption ty -> (
+    match Obj.tag o - Obj.first_non_constant_constructor_tag with
+    | 0 ->
+      let e =
```

Comment:
What does this do??

Reply:
OCaml variants are discriminated by tag ; since we only use non-constant constructors (they all have a single parameter, which may be unit), querying the OCaml tag gives us the index in the Catala constructor map (i.e. the order in which the constructors have been defined in the OCaml code).

Our constructor maps use comparison based on their UIDs, which are attributed in increasing order, at declaration time, so `ENone` will be 0 and `ESome` will be 1. (But yes, that's quite a few assumptions)